Lib/Protocols/IdFTP.pas

```diff
@@ -2742,18 +2764,45 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-
+  if FClientInfo.ClientName <> '' then begin
+    if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := 'Name='+FClientInfo.FClientName;
+      if FClientInfo.FClientVersion <> '' then begin
+        LClnt := LClnt + '; Version='+FClientInfo.FClientVersion;
+      end;
+      if FClientInfo.FVendor <> '' then begin
+        LClnt := LClnt + '; Vendor='+FClientInfo.FVendor;
+      end;
+      SendCmd('CSID ' + LClnt);  {do not localize}
+      LBuf := TrimRight(Self.LastCmdResult.Text.Text);
+      FServerInfo.CaseSensitive := True;
+      repeat
+         LClnt := Fetch(LBuf,';');
+         LValName := TrimLeft(Fetch(LClnt,'='));
+         case PosInStrArray(LValName,['Name','Version','Vendor','OS','OSVer','CaseSensitive']) of

```

Comment:
A `CSID` response can contain more name=value pairs than just the few you are looking for.  I would suggest providing a catch-all property to store any extra values. It may make more sense to just store all of the received values into a `TStringList` and then have a few property getters that pull from that list.